include/manif/impl/sgal3/SGal3Tangent_base.h

```diff
@@ -0,0 +1,635 @@
+#ifndef _MANIF_MANIF_SGAL3TANGENT_BASE_H_
+#define _MANIF_MANIF_SGAL3TANGENT_BASE_H_
+
+#include "manif/impl/sgal3/SGal3_properties.h"
+#include "manif/impl/tangent_base.h"
+#include "manif/impl/so3/SO3Tangent_map.h"
+#include "manif/impl/se3/SE3Tangent.h"
+
+namespace manif {
+
+//
+// Tangent
+//
+
+/**
+ * @brief The base class of the SGal3 tangent.
+ */
+template <typename _Derived>
+struct SGal3TangentBase : TangentBase<_Derived> {
+private:
+
+  using Base = TangentBase<_Derived>;
+  using Type = SGal3TangentBase<_Derived>;
+
+public:
+
+  MANIF_TANGENT_TYPEDEF
+  MANIF_INHERIT_TANGENT_API
+  MANIF_INHERIT_TANGENT_OPERATOR
+
+  using LinBlock = typename DataType::template FixedSegmentReturnType<3>::Type;
+  using AngBlock = typename DataType::template FixedSegmentReturnType<3>::Type;
+  using ConstLinBlock = typename DataType::template ConstFixedSegmentReturnType<3>::Type;
+  using ConstAngBlock = typename DataType::template ConstFixedSegmentReturnType<3>::Type;
+
+  using Base::data;
+  using Base::coeffs;
+
+protected:
+
+  using Base::derived;
+
+  MANIF_DEFAULT_CONSTRUCTOR(SGal3TangentBase)
+
+public:
+
+  MANIF_TANGENT_ML_ASSIGN_OP(SGal3TangentBase)
+
+  // Tangent common API
+
+  /**
+   * @brief Hat operator of SGal3.
+   * @return An element of the Lie algebra se_2_3.
+   * @note See Eq. (169).
+   */
+  LieAlg hat() const;
+
+  /**
+   * @brief Get the SGal3 element.
+   * @param[out] -optional- J_m_t Jacobian of the SGal3 element wrt this.
+   * @return The SGal3 element.
+   * @note This is the exp() map with the argument in vector form.
+   */
+  LieGroup exp(OptJacobianRef J_m_t = {}) const;
+
+  /**
+   * @brief This function is deprecated.
+   * Please considere using
+   * @ref exp instead.
+   */
+  MANIF_DEPRECATED
+  LieGroup retract(OptJacobianRef J_m_t = {}) const;
+
+  /**
+   * @brief Get the right Jacobian of SGal3.
+   */
+  Jacobian rjac() const;
+
+  /**
+   * @brief Get the left Jacobian of SGal3.
+   */
+  Jacobian ljac() const;
+
+  /**
+   * @brief Get the small adjoint matrix ad() of SGal3
+   * that maps isomorphic tangent vectors of SGal3
+   * @return
+   */
+  Jacobian smallAdj() const;
+
+  // SGal3Tangent specific API
+
+  //! @brief Get the linear velocity part.
+  LinBlock lin();
+  const ConstLinBlock lin() const;
+
+  //! @brief Get the angular part.
+  AngBlock ang();
+  const ConstAngBlock ang() const;
+
+  //! @brief Get the linear acceleration part
```

Comment:
this is `nu`, the linear velocity part.

Remember in IMU, we do refer to velocity and acceleration, but they are multiplied by Delta_t, becoming translation (aka position change) and velocity change.

Reply:
My bad for copy/pasting.